mage_ai/orchestration/db/models/schedules.py

```diff
@@ -212,16 +251,100 @@ def should_schedule(self) -> bool:
             current_execution_date = self.current_execution_date()
             if current_execution_date is None:
                 return False
+            # If there is a pipeline_run with an execution_date the same as the
+            # current_execution_date, then don’t schedule
             if not find(
                 lambda x: compare(
                     x.execution_date.replace(tzinfo=pytz.UTC),
                     current_execution_date,
                 ) == 0,
                 self.pipeline_runs
             ):
-                return True
+                if self.landing_time_enabled():
+                    if len(previous_runtimes) == 0:
+                        return True
+                    else:
+                        runtime = ceil(sum(previous_runtimes) / len(previous_runtimes))
+                        sd = ceil(stdev(previous_runtimes))
+                        if datetime.now(timezone.utc) >= current_execution_date - timedelta(
```

Comment:
I think it could cause duplicate pipeline runs being scheduled.

Reply:
I don’t think so unless we have 2 schedulers.